### ISSUE TITLE
Nextcloud users on map?

### ISSUE THREAD

**Issue by DocSlyper** [nextcloud/maps#344]
Hey there,
this app is very nice and useful - thank's for this :-)

While starting using the app it came to my mind that it might be cool to also have a standard layer that shows the location of the other users depending on their location information in the profile.
We want to use Nextcloud to stay in contact, share things, etc. But due to our job everyone is changing location frequently. Therefor it might be cool to see the current location of the other users on the map :-)

Not sure if this is possible, but normally the needed information to show user locations should be available in their profile... Not the "live" one, the one that every user can fills in the profile-field "Address" so that everyone also can decide to share or not - if not just leave empty.

Greetings from Germany
HW

**Comment by tacruc** [comment]
In general I like the Idea, but I have some Privacy concerns. 
A fast work arround would be to have your colleges in an shared addressbook.

**Comment by DocSlyper** [comment]
In the end the privacy problem might be higher in my opinion if one user creates a shared address book. In this case it is up to this colleague to take care of the privacy if one does not want to share his location. Or even worse if everyone can edit this shared address book ...
If the address field from the profile is used it is up to the user to share or not to share ...
So far I cannot see other important/mandatory use of this field, please correct me if I'm wrong. Therefor it's possible to leave it empty. Even if someone does not want to share the exact location, the city might be enough ;-)

**Comment by tacruc** [comment]
Sorry I misunderstood you first post. I understood some kind of live position depending on the device location.

Still I think that the shared address book is the faster workaround. If you want to have something which works directly.

**Comment by tacruc** [comment]
@jancborchardt how is the general discussion going with the handling of different sources of connected people. As I see it address books and user of an instant are just two different sources for contacts/people. Does the user of maps care if the marker is coming from an contact or from an address book?

**Comment by Spartachetto** [comment]
Isn't this related to #18 ?

What is described there is a possible use case for displaying users or contacts on a map.

Maybe the visualization could be ad hoc i.e.: if needed

**Comment by jancborchardt** [comment]
> Does the user of maps care if the marker is an other user or if it is from an address book?

Not really, I would agree with @DocSlyper that it’s nice to visualize other users on the same Nextcloud (i.e. colleagues) if they set an address and the privacy settings (which already exist) are set so that is visible.

**Comment by Spartachetto** [comment]
In addition to what stated before, it could maybe be useful to remember that it strongly depends on the nature of the Nextcloud instance. 

For example if the instance is put up by an employer and the address is the one of the workplace provided by the same employer, I imagine it would be difficult for the users to ask for "privacy"...

**Comment by jancborchardt** [comment]
Privacy settings are for users, and there are privacy options in Settings → Personal where you can limit the visibility of your address.

**Comment by bkil** [comment]
Real time location sharing and showing the track log of others can also be interesting for coordinating mapping parties
* https://github.com/osmandapp/Osmand/issues/7469

**Comment by SergioArbarviro** [comment]
Has there been any progress on this Issue ? 
I am in the same situation as the one described by @DocSlyper . I would like a map to show the address city of all NextCloud users that have decided to provide it to the 'Contacts' app and to display it. This means that there are no privacy issue here: a person that does not want to share his/her city of residence with the system, or who does not want to share it with the other users by default displays nothing. the display of the home city on the map only happens following an explicit and positive decision by the user. 
If the issue is still active, are there any plans to implement it, and if so, when?

**Comment by cherti** [comment]
What does it take to make the address book that is all users already show up in the maps application? to me it looks like everything is there, but given that I can already add all users to a shared address book and it would work, this meta-address book would just need to show up in the maps application

**Comment by SergioArbarviro** [comment]
@cherti I thank you very much for your suggestion. The difference is that the address book needs to be set up by hand, either by me adding each colleague individually, or by them having to actively register to it.
What I am looking for is that it be displayed by default, with no further action by anyone.

**Comment by cherti** [comment]
Nextcloud recently introduced a feature that all users on that Nextcloud instance are in an address book accessible to all users on that instance. Unfortunately, people in that address book don't show up for the map. Is it possible to get those users show up on the map as well? So far it has only worked if I download that address book and reupload it as a custom one. Which means the information is there, given that the reupload made people show up, but the built-in address book does not lead to pins showing up unfortunately.

**Comment by SergioArbarviro** [comment]
Hello @cherti 

I thank you very much for your very fast and very relevant comment.

I can indeed export the address book of all users, and re-import it to a another address book (which I called _"Community_Map"_)  :-)

I have however one  problem remaining.

I would like these contacts to be displayed on a **custom** map, not the default one. The reason is that I would like to share that map. When I create a new map, then the connection to the "_Community_Map_" address book is lost (it appears as empty) and its values are not displayed on the custom map. They are however well displayed on the "Default" map.

How can I make sure that the content of the "_Community_Map_" address book is displayed on a custom map (i.e. a map that is different from the "Default")?

**Comment by cherti** [comment]
My comments were actually less related to you but more to the developers of this plugin. :)

The ability to have the shared address book visible on the map would be a great addition to Nextcloud and this plugin and would make it even more valuable to our use of it. :)
